nixos/modules/services/misc/pufferpanel.nix

```diff
@@ -0,0 +1,401 @@
+    map bind compatPath.pathsToLink
+  );
+
+  stdenvDynamicLinker = pkgs.runCommand "pufferpanel-linker" { } ''
+    mkdir -p $out/lib
+    ln -s -t $out/lib "$(cat ${pkgs.stdenv.cc}/nix-support/dynamic-linker)"
+    ${lib.optionalString pkgs.stdenv.is64bit ''
+      ln -s $out/lib $out/lib64
+    ''}
+  '';
+in
+{
+  options.services.pufferpanel = {
+    enable = lib.mkEnableOption (lib.mdDoc ''
+      PufferPanel game management server.
+
+      The server can run as a panel with local daemon, a standalone daemon, or
+      a standalone panel.
+    '');
+
+    package = lib.mkPackageOptionMD pkgs "pufferpanel" { };
+
+    extraGroups = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      example = [ "docker" ];
+      description = lib.mdDoc ''
+        Additional groups for the systemd service.
+      '';
+    };
+
+    extraPackages = lib.mkOption {
+      type = lib.types.listOf lib.types.package;
+      default = [ ];
+      example = lib.literalExpression "[ pkgs.jre ]";
+      description = lib.mdDoc ''
+        Packages to add to the PATH environment variable. Both the {file}`bin`
+        and {file}`sbin` subdirectories of each package are added.
+      '';
+    };
+
+    compat = {
+      enable = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Enable best-effort compatibility layer for [templates] and binaries
+          downloaded by PufferPanel.
+
+          If set to `true`, packages specified in {option}`extraPackages` and
+          {option}`compat.dynamicLinker` (if not set to `null`) will be made
+          available at `/bin`, `/lib` (and `/lib64` on 64-bit systems) in the
+          service’s private filesystem namespace for corresponding package
+          subdirectories.
+
+          To cover requirements for most of the provided PufferPanel templates,
+          on `x86_64-linux` system, enable this option and use the following
+          list of packages as a reference:
+          ```Nix
+          services.pufferpanel.extraPackages =
+            with pkgs; [
+              coreutils
+              bash
+              gawk
+              gzip
+              gnutar
+              unzip
+              xz
+              wget
+              curl
+              git
+              mono
+              jre
+            ]
+            ++
+            map lib.getLib [
+              stdenv.cc.cc.lib # for libstdc++
+              zlib
+              icu
+              openssl
+              libgdiplus
+              curl
+              sqlite
+            ];
+          ```
+
+          Note, however, that this option should only be used as a last resort.
+          It is otherwise recommended to configure required packages via
+          {option}`extraPackages` with this option disabled.
+
+          [templates]: https://github.com/PufferPanel/templates
+        '';
+      };
+
+      dynamicLinker = lib.mkOption {
```

Comment:
Why do we need this option?

Reply:
Actually, no, it’s possible to set FHS-compatible PufferPanel wrapper (i.e. `buildFHSUserEnv`) using `package` option. I’ll document that and remove `compat` block. Requires loosening `RestrictNamespaces` to allow `mnt` and `user` (mount and user namespaces), but that shouldn’t be an issue.

Reply:
Done